feat(node): add SaaS git providers

grahamhar opened the pull request:
Adding in GitHub, BitBucket and GitLab as some of the commonly used git SaaS providers

gabriel-tessier commented:
@grahamhar 
Thanks for the PR actually the icons of your proposal are already registered here:

https://diagrams.mingrammer.com/docs/nodes/onprem#onpremvcs

But Bit bucket is not in the list, if you update your PR and add it in On Prem vcs, your PR can be merged.

grahamhar commented:
@gabriel-tessier  thanks for the review and sorry I didn't spot some of these were already in a different section.

I have moved as suggested but when I ran autogen.sh it looks like in previous PRs there may have been some manual changes as a few other changes have been picked up, I added them to my change let me know if if you want me to remove them.

gabriel-tessier commented:
@grahamhar 

Tanks for the changes, about the files from autogen you can remove them, include only the change you made.
Just 2 changes:
- Can you resize the icon to be minimum 256 as written in the contribution guide:
https://github.com/mingrammer/diagrams/blob/master/CONTRIBUTING.md

- Change the picture with an alpha or transparent background instead of white.

I will add the changes in the review to make more easy for other reviewer to validate.

gabriel-tessier reviewed (changes requested):
- [ ] Remove not concerned files from autgen
- [ ] Resize the icon to min 256
- [ ] Change Icon background to transparent
